frontend/catalyst/pennylane_extensions.py

```diff
@@ -1452,15 +1452,10 @@ def default_expand_fn(self, circuit, max_expansion=10):
         # decomposition is generally faster.
         # At the moment, bypassing decomposition for controlled gates will generally have a higher
         # success rate, as complex decomposition paths can fail to trace (c.f. PL #3521, #3522).
-
-        def _decomp_controlled_unitary(self, *_args, **_kwargs):
-            return qml.QubitUnitary(qml.matrix(self), wires=self.wires)
-
         def _decomp_controlled(self, *_args, **_kwargs):
-            return qml.QubitUnitary(qml.matrix(self), wires=self.wires)
+            return [qml.QubitUnitary(qml.matrix(self), wires=self.wires)]
 
         with Patcher(
-            (qml.ops.ControlledQubitUnitary, "compute_decomposition", _decomp_controlled_unitary),
```

Comment:
Is this class gone?

Reply:
The class is not gone, but `ControlledQubitUnitary.compute_decomposition` is not called even with a test. The one that gets called is `decomposition`. Perhaps something changed at some point?

Reply:
Shouldn't we patch `ControlledQubitUnitary.decomposition` then?

Reply:
It is already decomposed into a QubitUnitary: `    %45:3 = "quantum.unitary"(%35, %38, %41, %44) : (tensor<8x8xcomplex<f64>>, !quantum.bit, !quantum.bit, !quantum.bit) -> (!quantum.bit, !quantum.bit, !quantum.bit), since it is a subclass of `ControlledOp` which itself is a subclass of `Controlled`.
`

Reply:
Got it :)